llvm/lib/Transforms/InstCombine/InstCombineCasts.cpp

```diff
@@ -1467,6 +1478,15 @@ Instruction *InstCombinerImpl::visitSExt(SExtInst &Sext) {
     if (ComputeNumSignBits(X, 0, &Sext) > XBitSize - SrcBitSize)
       return CastInst::CreateIntegerCast(X, DestTy, /* isSigned */ true);
```

Comment:
Can we remove this code now? It should be redundant.

Reply:
Yes. After removal, no regression shown in dtcxzyw's benchmark. Thanks